### PR TITLE
Make working progress bar

### PR DESCRIPTION
**Pull request by shaavan**
- This PR makes the progress bar on the block sync page work.
- This was done by introducing a new private member in the NodeModel class.

**Screenshots:**

**Master**

Screenshot from 2021-12-28 11-38-51

**PR (at different instances)**

|60680 Blocks|65344 Blocks|69744 Blocks|
|----|----|----|
|Screenshot from 2021-12-28 11-35-27|Screenshot from 2021-12-28 11-35-34|Screenshot from 2021-12-28 11-35-41|

### CODE REVIEW

**Comment by shaavan** [comment]
Changes in following files:
- gui-qml.cflags
- gui-qml.config
- gui.qml.creator
- gui-qml.cxxflags
- gui-qml.files
- gui-qml.includes

They are created automatically by the Qt Creator.

If any of these changes are redundant or unwanted, kindly inform me, and I shall remove them in the next update of the PR.
Thank You.

**Comment by shaavan** [comment]
To test the PR, you can use the following command:

```
./src/qt/bitcoin-qt -signet -daemon -reindex
```

**Comment by jarolrod** [comment]
Please remove all of the Qt Creator specific files.

For now this is not exactly needed. You can leave this open for now, I'll include this as part of my onboarding PR with some changes.

**Comment by shaavan** [comment]
Updated from fa36f92b80ff1b8359bdc4a49f280595e31c1da0 to ff40e069ef1947a7354afafa6d49dceb6ebeb5f5 ([pr99.01](https://github.com/shaavan/gui-qml/tree/pr99.01) -> [pr99.02](https://github.com/shaavan/gui-qml/tree/pr99.02))
Addressed @jarolrod comments

Changes:

- Removed Qt creator-specific changes from the PR.

> I'll include this as part of my onboarding PR with some changes.

Sound great!

**Comment by promag** [comment]
ACK exposing the verification progress property so it can be binded elsewhere.

File: src/qml/nodemodel.cpp [hunk 1]
```diff
@@ -23,6 +23,14 @@ void NodeModel::setBlockTipHeight(int new_height)
     }
 }
 
+void NodeModel::setVerificationProgress(double new_progress)
+{
+    if(new_progress != m_verification_progress) {
```
Comment (hebasto) [inline src/qml/nodemodel.cpp:28]: style:
```suggestion
    if (new_progress != m_verification_progress) {
```

File: src/qml/nodemodel.cpp [hunk 2]
```diff
@@ -37,13 +45,17 @@ void NodeModel::initializeResult([[maybe_unused]] bool success, interfaces::Bloc
...
     // TODO: Handle the `success` parameter,
     setBlockTipHeight(tip_info.block_height);
+    setVerificationProgress(tip_info.verification_progress);
 }
 
 void NodeModel::ConnectToBlockTipSignal()
 {
     assert(!m_handler_notify_block_tip);
+    assert(!m_verification_progress);
+
```
Comment (hebasto) [inline src/qml/nodemodel.cpp:55]: Why this assertion?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

Could you please split changes in `src/qml/pages/stub.qml` into a separated commit? Expecting it will improve maintainability (e.g., during syncing with the main repo). For example, see b6280a9c022ae5ac0c5c093ab77d86fd5bbbc652 and 1441d5be79f161e7864bf5493b5ed68c51731eec.

Also it'd nice if QML-only-stuff commits have a `qml:` prefix.

**Comment by shaavan** [comment]
Updated from ff40e069ef1947a7354afafa6d49dceb6ebeb5f5 to da06542c7968077d544906ef3e6fbfc56416dcca ([pr99.02](https://github.com/shaavan/gui-qml/tree/pr99.02) -> [pr99.03](https://github.com/shaavan/gui-qml/tree/pr99.03), [diff](https://github.com/shaavan/gui-qml/compare/pr99.02..pr99.03))
Addressed @hebasto comments.

Changes:

- Fixed styling
- Remove unnecessary assert.

File: src/qml/nodemodel.cpp (hunk 2 above)
Comment (shaavan) [reply src/qml/nodemodel.cpp]: That should not have been there. I was doing some trying and testing but forgot to remove this statement when I was done.
Thanks for catching this.
Fixed in da06542c7968077d544906ef3e6fbfc56416dcca

File: src/qml/pages/stub.qml [hunk 3]
```diff
@@ -35,7 +35,7 @@ ApplicationWindow {
         ProgressIndicator {
             id: indicator
             Layout.fillWidth: true
-            progress: 0.666
+            progress: nodeModel.verificationProgress
```
Comment (hebasto) [inline src/qml/pages/stub.qml:38]: Should we drop the following `background: MouseArea` now?

**Review by hebasto — COMMENTED** [review]
> Also it'd nice if QML-only-stuff commits have a `qml:` prefix.

Please add "qml: " to the beginning of your commit message.

**Comment by hebasto** [comment]
@shaavan 

And please split in two commits (https://github.com/bitcoin-core/gui-qml/pull/99#pullrequestreview-844475025).

**Comment by shaavan** [comment]
Updated from da06542c7968077d544906ef3e6fbfc56416dcca to bcea86c33ed98dfe11c6c7a8153032350c9165f8 ([pr99.03](https://github.com/shaavan/gui-qml/tree/pr99.03) -> [pr99.04](https://github.com/shaavan/gui-qml/tree/pr99.04), [diff](https://github.com/shaavan/gui-qml/compare/pr99.03..pr99.04))
Addressed @hebasto comments.

Changes:

- Split PR into two commits. Added `qml:` prefix to each of them
- Removed temporary mouse click event from the progress bar.

**Review by hebasto — APPROVED** [review]
ACK bcea86c33ed98dfe11c6c7a8153032350c9165f8

**Review by promag — COMMENTED** [review]
ACK bcea86c33ed98dfe11c6c7a8153032350c9165f8

**Pull request merged**